### PR TITLE
init copy default signing config

### PR DESCRIPTION
**Pull request by ilevyor**
*Issue #, if available:*

*Description of changes:*
Add init_copy_default_signing_config function

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by TingDaoK — COMMENTED** [review]
how you want to use it? If you want to get the config from the client, it should be from `aws_cached_signing_config_aws`, right? This just makes no sense, you can just use `=` ti copy each member.

File: source/s3_meta_request.c [hunk 1]
```diff
@@ -610,6 +610,20 @@ static void s_s3_meta_request_sign_request(
     meta_request->vtable->sign_request(meta_request, request, on_signing_complete, user_data);
 }
 
+static bool s_is_put_request(const struct aws_s3_meta_request *meta_request) {
+    if (meta_request->type == AWS_S3_META_REQUEST_TYPE_PUT_OBJECT) {
+        return true;
+    }
+    if (meta_request->type == AWS_S3_META_REQUEST_TYPE_DEFAULT) {
+        struct aws_byte_cursor method;
+        int err = aws_http_message_get_request_method(meta_request->initial_request_message, &method);
```
Comment (bretambrose) [inline source/s3_meta_request.c:619]: Feels like you could simplify this to something like:

```
if (!aws_http_message_get_request_method(...)) {
   return aws_byte_cursor_eq(...);
}
```
and let the fall through the hit the return false;

File: source/s3_meta_request.c [hunk 2]
```diff
@@ -663,6 +677,10 @@ void aws_s3_meta_request_sign_request_default(
         return;
     }
 
+    if (meta_request->checksum_algorithm != AWS_SCA_NONE && s_is_put_request(meta_request) &&
```
Comment (DmitriyMusatkin) [inline source/s3_meta_request.c:680]: nit: one condition per line would be more readable.
also comment on why it needs to be done would be helpful

**Review by bretambrose — APPROVED** [review]

**Review by TingDaoK — APPROVED** [review]

File: source/s3_meta_request.c (hunk 2 above)
Comment (ilevyor) [reply source/s3_meta_request.c:680]: formatter doesn't like separating conditions here.

**Pull request merged**